### PR TITLE
Disabled Continue Movement button

### LINKED ISSUE #107: Disable Continue Movement button for a hero which doesn't have chosen path
We must disable **Continue Movement** button if chosen hero doesn't have path to move. Here is a screenshot of current application:
image

and here is a screenshot of correct behavior:
image

### PR DESCRIPTION
**Pull request by kgorecki**
Disabled Continue Movement button for a hero which doesn't have chosen path. Resolves #107

### CODE REVIEW

**Comment by kgorecki** [comment]
@ihhub Travis complained that nullptr was not declared on Linux builds. Should I change it to NULL or should we update to C++11?

**Comment by ihhub** [comment]
@kgorecki the project doesn't support C++11. Please use `NULL` instead.

File: src/fheroes2/gui/interface_buttons.cpp [hunk 1]
```diff
@@ -97,6 +97,11 @@ void Interface::ButtonsArea::Redraw(void)
 
     SetButtonStatus();
 
+	Heroes * currentHero = GetFocusHeroes();
```
Comment (ihhub) [inline src/fheroes2/gui/interface_buttons.cpp]: Please, move this code into `SetButtonStatus()` function.

File: src/fheroes2/gui/interface_events.cpp [hunk 2]
```diff
@@ -55,6 +55,8 @@ void Interface::Basic::ShowPathOrStartMoveHero(Heroes* hero, s32 dst_index)
         DEBUG(DBG_GAME, DBG_TRACE, hero->GetName() << ", route: " << path.String());
         gameArea.SetRedraw();
         cursor.SetThemes(GetCursorTileIndex(dst_index));
+        Interface::Basic & I = Interface::Basic::Get();
```
Comment (ihhub) [inline src/fheroes2/gui/interface_events.cpp]: Please group these 2 lines into 1 code line.

File: src/fheroes2/gui/interface_buttons.cpp (hunk 1 above)
Comment (kgorecki) [reply src/fheroes2/gui/interface_buttons.cpp]: You're absolutely right, I feel ashamed...

File: src/fheroes2/gui/interface_events.cpp (hunk 2 above)
Comment (kgorecki) [reply src/fheroes2/gui/interface_events.cpp]: Done

File: src/fheroes2/gui/interface_buttons.cpp [hunk 3]
```diff
@@ -206,5 +206,10 @@ int Interface::ButtonsArea::QueueEventProcessing(void)
 
 void Interface::ButtonsArea::SetButtonStatus()
 {
-    buttonMovement.SetDisable(GetFocusHeroes() == NULL);
```
Comment (ihhub) [inline src/fheroes2/gui/interface_buttons.cpp]: You removed previous statement for `GetFocusHeroes() == NULL` which disables the button for castles :)

**Comment by ihhub** [comment]
@kgorecki thank you very much for fixing this issue! :)

File: src/fheroes2/gui/interface_buttons.cpp (hunk 3 above)
Comment (kgorecki) [reply src/fheroes2/gui/interface_buttons.cpp]: Yeah, you're right :}
Just a quick thought to your commit - what about splitting setting isMovementButtonDisabled into 2 lines? It's quite hard to read line which is over 120 chars.

Comment (ihhub) [reply src/fheroes2/gui/interface_buttons.cpp]: 120 characters is not big with nowadays monitors. Many commercial projects have limit in 180-200 characters.

Comment (kgorecki) [reply src/fheroes2/gui/interface_buttons.cpp]: Makes sense.

**Pull request merged**